benchexec/tools/coveriteam-verifier-validator.py

```diff
@@ -50,14 +51,16 @@ def determine_result(self, run):
         will print out the produced aftifacts.
         If more than one dict is printed, the first matching one.
         """
-        for line in run.output:
+        verdict = None
+        verdict_regex = re.compile(r"'verdict': '([a-zA-Z\(\)\ ]*)'")
+        for line in reversed(run.output):
             line = line.strip()
-            if "verdict" in line:
+            verdict_match = verdict_regex.search(line)
+            if verdict_match and verdict is None:
                 # CoVeriTeam outputs benchexec result categories as verdicts.
-                try:
-                    d = ast.literal_eval(line)
-                    if isinstance(d, dict):
-                        return d.get("verdict", result.RESULT_ERROR)
-                except SyntaxError:
-                    pass
-        return result.RESULT_ERROR
+                verdict = verdict_match.group(1).lower()
```

Comment:
would returning the verdict at this point be better? or do you see a case where an exception is logged but still a verdict is produced later, and we are interested in the exception instead of the verdict?

Reply:
Good point, that's exactly the case!
If `--debug` is used, the debug log may still print the output artifacts, which includes an (invalid) verdict.
Since we go through the list from bottom to top, this may be falsely identified as the final verdict if we don't continue.

But you got me thinking, maybe we should ignore lines that start with DEBUG, even if they match the regular expression for a verdict?
I'll try to find an output log that matches my above description.

Reply:
Sorry for the long wait. I couldn't create a log that creates an exception and prints a verdict afterwards, but I guess the current behavior would still be more robust in case such a case ever occurs. Are you ok with that?

*Just for completeness:*

We also can't ignore DEBUG lines, unfortunately, because CoVeriTeam logs messages with newlines.

```
2021-01-04 14:40:45,709 DEBUG Actor cpachecker-predicate-no-cegar acted on artifacts:
 {'program': '../../sv-benchmarks/c/seq-mthreaded/rekcba_ctm.3-2.c', 'spec': '../../sv-benchmarks/c/properties/unreach-call.prp', 'predicates': ''} 
and produced artifacts:
 {'verdict': 'false(unreach-call)', 'witness': '', 'predicates': 'cvt-output/4cfccc45-4f98-4da1-8ace-481f9f79fc87/cpachecker-predicate-no-cegar/c19d57ab-f16f-4416-a6e5-9ff67e55e128/output/predmap.txt'}
```

So that idea didn't help.